README.rst

```diff
@@ -284,6 +284,9 @@ extended in git-config(1) files::
   [gitpublishprofile "default"]
   suppresscc = all            # do not auto-cc people
 
+Additionally, if the default profile is empty and no --profile command-line
+option was given, git-publish will use the first profile it finds.
+
```

Comment:
> but the commit seems to purposely defeat this

Not really. If the user has specified their defaults, i.e. a profile named "default" exists, nothing changes.

I'm just trying to find a way to make it more intuitive. My first encounter with git-publish didn't end well because .gitpublish had one non-default profile and an incorrect comment about the first profile always being used by default. Running git-publish without -p then silently picked up the empty "default" and didn't do what I expected it to. In all honesty, the way "default" is designed is unfortunate.

> Why not adjust .gitpublish to use "default" as the profile name if it should be used by default?

Would that work for power users who may have their own "default" profile defined outside of .gitpublish?

As an alternative to this patch I'd settle for:

1) The logic proposed here but using the non-default profile by default only if it's the only one. I.e. "if len(lines) == 1:" instead of "if len(lines):"
So - if there is only one profile, then it is used. Regardless of name. Hard to argue with that! :)

2) If not even that is acceptable, then at least print the profile being used and warn if it is "default" and it's not defined. Because that's almost certainly not what the user wants.